### PR TITLE
Refactor backend eslint

### PR DESCRIPTION
**Pull request by Sidhant268**
Contents:
Refactored the backend code base
Added eslint support
Partitioned Cost Price Service into two different services
Added service layer for the missing ones
Removed unused and eslint error code base

Please review and merge

### CODE REVIEW

File: server/services/costPriceChangeService.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+const ApiConfig = require("../config/api.config");
+const AnalyticsService = require("./analyticsService");
+
+const getCostPriceChangeRequestsById = async (eventId) => {
+  try {
+    const cpChangeRequests = await CostPriceChangeRequests.find({
+      event_id: eventId,
+    });
+    logger.info(
+      `Cost price change requests data fetched for eventId: ${eventId}`
```
Comment (arun-bonito) [inline server/services/costPriceChangeService.js:16]: Please use " instead of ` for consistency.

File: server/services/costPriceChangeService.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    if (purchasePriceAppListLinesResponse.value) {
+      console.log(purchasePriceAppListLinesResponse.value[0]);
+      return purchasePriceAppListLinesResponse.value[0];
+    } else {
+      throw new Error("Purchase Price App List lines not present");
+    }
+  } catch (error) {
+    if (error.response) {
+      console.error(
+        `Error response status: ${error.response.status} and Error response data:`,
```
Comment (arun-bonito) [inline server/services/costPriceChangeService.js:87]: Please use " instead of ` for consistency.

File: server/services/costPriceChangeService.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+          etag: doc.etag,
+        },
+      },
+    },
+  }));
+
+  try {
+    const result = await CostPriceChangeRequests.bulkWrite(bulkOps);
+    console.log(
+      `Updated ${result.modifiedCount} cost price change requests documents`
```
Comment (arun-bonito) [inline server/services/costPriceChangeService.js:156]: Use "

File: server/services/costPriceChangeService.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    const payload = { DirectUnitCost: newCostPrice };
+    const apiResponse = await axios.patch(apiUrl, payload, {
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+        "Content-Type": "application/json",
+        "If-Match": `${etag}`,
+      },
+    });
+
+    if (apiResponse.status == 200) {
```
Comment (arun-bonito) [inline server/services/costPriceChangeService.js:189]: Change to return (apiResponse.status == 200)

File: server/services/costPriceService.js [hunk 5]
```diff
@@ -1,162 +1,120 @@
...
-    } else {
-      throw new Error("Purchase Price App List lines not present");
-    }
+    logJson(response.data);
+    logger.info("Vendor Details fetched!");
+    return response.data;
   } catch (error) {
-    if (error.response) {
-      console.error(
-        `Error response status: ${error.response.status} and Error response data:`,
```
Comment (arun-bonito) [inline server/services/costPriceService.js:43]: Use " instead of `

File: server/services/costPriceService.js [hunk 6]
```diff
@@ -1,162 +1,120 @@
...
 
-  try {
-    const result = await CostPriceChangeRequests.bulkWrite(bulkOps);
-    console.log(
-      `Updated ${result.modifiedCount} cost price change requests documents`
+    const eventData = document.event_details.find((event) =>
+      event.data.some((data) => data.event_no === event_no)
     );
-    logger.info(
-      `Successfully updated the old cost price and erp sku code in change requests for event id: ${cpChangeRequests.event_id}`
```
Comment (arun-bonito) [inline server/services/costPriceService.js:115]: Use " instead of `

File: server/services/sellingPriceService.js [hunk 7]
```diff
@@ -1,11 +1,9 @@
 const SellingPriceRequests = require("../models/sellingPriceRequests");
 const costPriceChangeRequests = require("../models/costPriceChangeRequests");
-const VendorRequests = require("../models/vendorRequests");
 const AnalyticsService = require("../services/analyticsService");
 const User = require("../models/user");
 const { logger, logJson } = require("../utils/logger");
```
Comment (arun-bonito) [inline server/services/sellingPriceService.js:5]: Remove logJson if not needed.

File: server/services/sellingPriceService.js [hunk 8]
```diff
@@ -179,8 +177,10 @@ const calculateSellingPrice = async () => {
     const startDate = new Date(currentYear, currentMonth, 1);
     const endDate = new Date(currentYear, currentMonth + 1, 0);
     endDate.setHours(23, 59, 59, 999);
-    logger.info(`looking for CPChangeRequests within ${startDate} and ${endDate}`);
-    
+    logger.info(
+      `looking for CPChangeRequests within ${startDate} and ${endDate}`
```
Comment (arun-bonito) [inline server/services/sellingPriceService.js:181]: Use " instead of `

File: server/services/sellingPriceService.js [hunk 9]
```diff
@@ -229,11 +229,11 @@ const calculateSellingPrice = async () => {
...
-        _id: { $in: filteredCostPriceChangeRequests.map(req => req._id) },
+        _id: { $in: filteredCostPriceChangeRequests.map((req) => req._id) },
       },
       { $set: { status: "SP_CALCULATED" } }
     );
-    logger.info("Successfully calulated selling prices.")
+    logger.info("Successfully calulated selling prices.");
     return true;
   } catch (error) {
     logger.error("Error while caluclating selling prices:", error);
```
Comment (arun-bonito) [inline server/services/sellingPriceService.js:239]: Add throw new error ("Error while caluclating selling prices:", error) inside the catch block.

**Review by arun-bonito — CHANGES_REQUESTED** [review]
Added comments, please check.

File: server/services/costPriceChangeService.js (hunk 1 above)
Comment (Sidhant268) [reply server/services/costPriceChangeService.js:16]: ` is used when we are adding a variable dynamically. We cannot read the ${eventId} by using ".

File: server/services/sellingPriceService.js (hunk 7 above)
Comment (Sidhant268) [reply server/services/sellingPriceService.js:5]: logJson is used in the following file

File: server/services/sellingPriceService.js (hunk 9 above)
Comment (Sidhant268) [reply server/services/sellingPriceService.js:239]: After logging there is a throw block.

Comment (arun-bonito) [reply server/services/sellingPriceService.js:239]: Okay, saw it now. GTG.

**Comment by Sidhant268** [comment]
Removed the redundant if else clause in the sellingPriceService.js file

Kindly review and Merge @arun-bonito

**Pull request merged**